arkane/statmech.py

```diff
@@ -253,17 +253,13 @@ def load(self, pdep=False):
 
         try:
             linear = local_context['linear']
-            symfromlog = False
         except KeyError:
             externalSymmetry = None
```

Comment:
(not introduced in this PR): Could you also remove `externalSymmetry = None` here? The next try except block sets `externalSymmetry` to either the user input or None anyway. Let's fix it while we're here.
I'm thinking:
```
try:
    linear = local_context['linear']
except KeyError:
    logging.error('some helpful message')
    raise
```
I have a function for automatically determining whether a species is linear (using xyz) + tests in ARC, I'll migrate it from ARC here if `linear` isn't given by the user instead of raising an error.